src/OSPSuite.Core/Domain/Model.cs

```diff
@@ -35,39 +37,32 @@ public IContainer Neighborhoods
                Root.RemoveChild(_neighborhoods);
             }
             _neighborhoods = value;
-            if (Root != null)
-            {
-               Root.Add(_neighborhoods);
-            }
+            Root?.Add(_neighborhoods);
          }
-         get { return _neighborhoods; }
+         get => _neighborhoods;
       }
 
       public double? MolWeightFor(IQuantity quantity)
       {
-         if (Root == null)
+         if (Root == null || quantity == null)
```

Comment:
This method was too restrictive I believe. It was returning the Molweight of the associated quantity only for `MoleculeAmount` AND `Observer`. But it should be possible to retrieve the MolWeight for all other types of quantity (I think the last remaining one is `Parameter`, but no need to be explicit about that here)